src/sync/app.hpp

```diff
@@ -38,23 +44,32 @@ class RemoteMongoClient;
 /// and writing on the database.
 ///
 /// You can also use it to execute [Functions](https://docs.mongodb.com/stitch/functions/).
-class App : public AuthRequestClient, public AppServiceClient {
+class App : std::enable_shared_from_this<App>, public AuthRequestClient, public AppServiceClient {
 public:
     struct Config {
         std::string app_id;
         GenericNetworkTransport::NetworkTransportFactory transport_generator;
-        realm::util::Optional<std::string> base_url;
-        realm::util::Optional<std::string> local_app_name;
-        realm::util::Optional<std::string> local_app_version;
-        realm::util::Optional<uint64_t> default_request_timeout_ms;
+        util::Optional<std::string> base_url;
+        util::Optional<std::string> local_app_name;
+        util::Optional<std::string> local_app_version;
+        util::Optional<uint64_t> default_request_timeout_ms;
     };
 
     App(const Config& config);
+    App() = default;
```

Comment:
Why do we need this constructor?

Reply:
You tell me 😝 
https://github.com/realm/realm-object-store/pull/942/files#diff-08c2f8b03a8e4ef38c589102149ed023R59

Reply:
I was but a young padawan back then.

Reply:
removed